tests/unit/src/test/scala/tests/CodeLensesSlowSuite.scala

```diff
@@ -0,0 +1,41 @@
+package tests
+import scala.concurrent.Future
+
+object CodeLensesSlowSuite extends BaseSlowSuite("codeLenses") {
+  testAsync("run") {
+    for {
+      _ <- server.initialize(
+        """|/metals.json
+           |{
+           |  "a": { }
+           |}
+           |
+           |/a/src/main/scala/Main.scala
+           |object Main {
+           |  def main(args: Array[String]): Unit = {}
+           |}""".stripMargin
+      )
+      _ <- verifyCodeLenses(
+        "a/src/main/scala/Main.scala",
+        """<<metals-scala-run>>
```

Comment:
Nice!

Reply:
but not ideal :(
it could also cover the UI name and command arguments like in:
`<<name shown on the UI | command(arg1, arg2, arg3)>>`
e.g. `<<run|metals-scala-run(file://foo/Main.scala,Main)>>`

but it would be much more cluttered (not to mention, that it would be ugly when expecting a `uri` argument.

Reply:
URIs can be pretty-printed to relative files. I think it might be best to to display the text that appears in the editor and ignore the command ID and arguments (since they're implementation details). We care most about how the code lens appears in the UI, we should test the functionality works as expected separately.